test/integration/controller/jobs/kubeflow/kubeflowjob.go

```diff
@@ -0,0 +1,304 @@
+		return ok
+	}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+	ginkgo.By("checking the workload is updated with new count")
+	gomega.Eventually(func() bool {
+		if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+			return false
+		}
+		return createdWorkload.Spec.PodSets[1].Count == newParallelism
+	}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+	gomega.Expect(createdWorkload.Status.Admission).Should(gomega.BeNil())
+
+	ginkgo.By("checking the job is unsuspended and selectors added when workload is assigned again")
+	admission = testing.MakeAdmission(clusterQueue.Name).PodSets(CreateMasterAndWorkerPodSet(createdWorkload, "on-demand", "spot")...).Obj()
+	gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+	util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+	gomega.Eventually(func() bool {
+		if err := k8sClient.Get(ctx, lookupKey, createdJob.Obj()); err != nil {
+			return false
+		}
+		return !*createdJob.RunPolicy().Suspend
+	}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+	gomega.Expect(len(createdJob.ReplicaSpecs()[master].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+	gomega.Expect(createdJob.ReplicaSpecs()[master].Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
+	gomega.Expect(len(createdJob.ReplicaSpecs()[worker].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+	gomega.Expect(createdJob.ReplicaSpecs()[worker].Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(spotFlavor.Name))
+	gomega.Consistently(func() bool {
+		if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+			return false
+		}
+		return len(createdWorkload.Status.Conditions) == 2
+	}, util.ConsistentDuration, util.Interval).Should(gomega.BeTrue())
+
+	ginkgo.By("checking the workload is finished when job is completed")
+	createdJob.JobStatus().Conditions = append(createdJob.JobStatus().Conditions,
+		kftraining.JobCondition{
+			Type:               kftraining.JobSucceeded,
+			Status:             corev1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+		})
+	gomega.Expect(k8sClient.Status().Update(ctx, createdJob.Obj())).Should(gomega.Succeed())
+	gomega.Eventually(func() bool {
+		err := k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+		if err != nil || len(createdWorkload.Status.Conditions) == 2 {
+			return false
+		}
+
+		return apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadFinished)
+	}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+}
+
+// CreateMasterAndWorkerPodSet creates a master and worker pod sets.
+func CreateMasterAndWorkerPodSet(createdWorkload *kueue.Workload, masterCpu, workerCpu kueue.ResourceFlavorReference) []kueue.PodSetAssignment {
+	return []kueue.PodSetAssignment{
```

Comment:
I think we shouldn't use this commonize function since not all kubeflow frameworks use the `master` and `worker` roles. For example, the MXJob uses `scheduler`, `server`, and `worker`.

Also, this commonize seems to violate the DAMP principle. I would suggest to refer to the SIG scheduling contribution guide: https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md#technical-and-style-guidelines

Reply:
I extracted it to a function because it appears 2 times in test. Now the function works for all kubeflow jobs, is it better?

Reply:
I'm ok with the updated function. It looks like keeps readability.